app/models/user.rb

```diff
@@ -1,6 +1,34 @@
 class User < ActiveRecord::Base
+  mount_uploader :avatar, AvatarUploader
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+  # :authentication_keys => [:login]
+  attr_accessor :login
+
+  def self.find_for_database_authentication(warden_conditions)
+  # def self.find_first_by_auth_conditions(warden_conditions)
```

Comment:
Not sure what hound's message means.  Github seems to be reading spacing differently than atom, and I can't seem to fix it.